pyccel/codegen/printing/ccode.py

```diff
@@ -668,6 +698,11 @@ def _print_FunctionDef(self, expr):
         decs += [Declare(i.dtype, i) for i in self._additional_declare]
         decs  = '\n'.join(self._print(i) for i in decs)
         self._additional_declare.clear()
+
+        if len(expr.results) > 1:
+            self._additional_args = [a.clone(name = self._parser.get_new_name('tmp_ret'), is_pointer =True) for a in expr.results]
```

Comment:
When you clone you create a new object which can break references to this variable elsewhere (they will not be updated to be pointers). How are you working round this problem?

Reply:
I have just realised that we cannot do operations with these variables so this is not a problem :+1: